go.mod

```diff
@@ -72,7 +75,11 @@ replace (
 	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20200410145947-61e04a5be9a6
 	k8s.io/kube-proxy => k8s.io/kube-proxy v0.18.5
 	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.18.5
-	k8s.io/kubectl => k8s.io/kubectl v0.18.5
+	// v0.18.x of kubectl vendored a really old version of github.com/docker/docker/pkg/term/windows, which
+	// vendored github.com/Sirupsen/logrus (with a capital S), hence go.mod won't parse. Using v0.19.0-alpha.1
```

Comment:
> v0.18.x of kubectl vendored a really old version of github.com/docker/docker/pkg/term/windows, which vendored github.com/Sirupsen/logrus

can you point to where you are seeing that? k8s.io/kubectl uses [be7ac8be2ae0 of github.com/docker/docker](https://github.com/kubernetes/kubectl/blob/release-1.18/go.mod#L14) and [has no transitive dependencies on the upper-case Sirupsen module](https://github.com/kubernetes/kubectl/blob/release-1.18/go.sum#L206-L208)

That version of docker uses github.com/sirupsen/logrus correctly as far as I can tell:
* https://github.com/moby/moby/blob/be7ac8be2ae0/vendor.conf#L14
* https://github.com/moby/moby/tree/be7ac8be2ae072032a4005e8f232be3fc57e4127/vendor/github.com/sirupsen/logrus
* searching the docker/docker repo at be7ac8be2ae0 for `Sirupsen` returns no results

Reply:
It looks like docker/docker has some dual versioning tags that confused go.mod version resolution... k8s.io/kubectl 1.18.x requires `github.com/docker/docker@v0.7.3-0.20190327010347-be7ac8be2ae0` which means "use sha be7ac8be2ae0, which was added when v0.7.3 was the closest earlier tag"

It looks like docker/docker has added more version tags in the meantime, so now doing `go get github.com/docker/docker@be7ac8be2ae0` resolves to `v17.12.0-ce-rc1.0.20190327010347-be7ac8be2ae0+incompatible`

`go mod graph | grep docker/docker` shows this:
```
github.com/weaveworks/footloose@v0.0.0-20200609124411-8f3df89ea188 github.com/docker/docker@v1.13.1
```

because github.com/docker/docker@v1.13.1 is semantically larger than `github.com/docker/docker@v0.7.3-0.20190327010347-be7ac8be2ae0`, v1.13.1 is being used, and *that* is the problematic version of docker:

https://github.com/moby/moby/blob/v1.13.1/vendor.conf#L5

I was able to run the following successfully after checking out this branch and reverting the go.mod and go.sum change:

```shell
go get k8s.io/kubectl@v0.18.5`
go get github.com/docker/docker@be7ac8be2ae0
go mod tidy
go build ./cmd/wksctl
```

Reply:
Oh I see. Updated now and it's working.

You're amazing :star:! Thanks a lot for digging into it :heart: